src/main.cpp

```diff
@@ -40,22 +41,25 @@ int main(int argc, char **argv) {
   ros::Rate update_rate(update_rate_hz);
   ros::Time ts = ros::Time::now();
   while (!shouldShutdown) {
-    forqueSensorHW.update();
+    errorShutdown = !forqueSensorHW.update();
     ros::Duration d = ros::Time::now() - ts;
     ts = ros::Time::now();
     cm.update(ts, d);
     update_rate.sleep();
+    shouldShutdown = errorShutdown;
   }
 
   // wait for a bit, so controller_manager can unload controllers
-  ros::Duration shutdownDuration(1.5);
-  ros::Time shutdownBegin = ros::Time::now();
-  while (ros::Time::now() - shutdownBegin < shutdownDuration) {
-    ros::Duration d = ros::Time::now() - ts;
-    ts = ros::Time::now();
-    cm.update(ts, d);
+  if(!errorShutdown) {
```

Comment:
I think this might be cleaner if it's just

```cpp
while (!shouldShutdown) {
  if (!forqueSensorHW.update())
    break;
  // the rest of the loop stuff
}

spinner.stop();
```

Then I think we can skip setting `shouldShutdown` above.

As a side note, I think as currently written it would sleep for the update rate. I don't think that's desired, although I also don't think it matters much.